mesonbuild/compilers/c.py

```diff
@@ -168,21 +168,21 @@ def get_include_args(self, path, is_system):
     def get_std_shared_lib_link_args(self):
         return ['-shared']
 
-    def get_library_dirs_real(self):
-        env = os.environ.copy()
-        env['LC_ALL'] = 'C'
-        stdo = Popen_safe(self.exelist + ['--print-search-dirs'], env=env)[1]
-        paths = []
-        for line in stdo.split('\n'):
-            if line.startswith('libraries:'):
-                libstr = line.split('=', 1)[1]
-                paths = [os.path.realpath(p) for p in libstr.split(':') if os.path.exists(os.path.realpath(p))]
-        return paths
-
-    def get_library_dirs(self):
-        key = tuple(self.exelist)
+    def get_library_dirs_real(self, env):
+        code=''
+        extra_args = ['--print-search-dirs']
+        with self._build_wrapper(code, env, extra_args, None, 'compile', True) as p:
+            paths = []
+            for line in p.stdo.split('\n'):
+                if line.startswith('libraries:'):
+                    libstr = line.split('=', 1)[1]
+                    paths = [os.path.realpath(p) for p in libstr.split(':') if os.path.exists(os.path.realpath(p))]
+            return paths
+
+    def get_library_dirs(self, env=None):
+        key = tuple(self.exelist) # todo include environment here
```

Comment:
Help requested as to how to manage the environment here. Is it possible to hash it in with the compiler to save the results for later?

Reply:
The `Environment` object is hashable, afair